llms.md

```diff
@@ -0,0 +1,1039 @@
+Alternatively, multiple `#[table(...)]` attributes can be applied directly to a single struct definition. This maps the same underlying type definition to multiple distinct tables:
+
+```rust
+use spacetimedb::{table, SpacetimeType, Identity, Timestamp};
+
+// Define the core data structure once
+// Apply multiple #[table] attributes to map it to different tables
+#[derive(SpacetimeType, Clone, Debug)] // Mark as type if used elsewhere
+#[table(name = "logged_in_players", public)]
+#[table(name = "players_in_lobby", public)]
+pub struct PlayerSessionData {
+    #[primary_key]
+    player_id: Identity,
+    #[unique]
+    #[auto_inc]
+    session_id: u64,
+    last_activity: Timestamp,
+}
+
+// Reducers would interact with the specific table handles:
+// E.g., ctx.db.logged_in_players().insert(PlayerSessionData { ... });
+// E.g., let lobby_player = ctx.db.players_in_lobby().player_id().find(&some_id);
+```
+
+##### Browsing Generated Table APIs
+
+The `#[table]` macro generates specific accessor methods based on your table definition (name, fields, indexes, constraints). To see the exact API generated for your tables:
+
+1.  Run `cargo doc --open` in your module project directory.
+2.  This compiles your code and opens the generated documentation in your web browser.
+3.  Navigate to your module's documentation. You will find:
+    *   The struct you defined (e.g., `PlayerState`).
+    *   A generated struct representing the table handle (e.g., `player_state__TableHandle`), which implements `spacetimedb::Table` and contains methods for accessing indexes and unique columns.
+    *   A generated trait (e.g., `player_state`) used to access the table handle via `ctx.db.{table_name}()`.
+
+Reviewing this generated documentation is the best way to understand the specific methods available for interacting with your defined tables and their indexes.
+
+#### Defining Reducers
+
+Reducers are functions that modify table data atomically. They are annotated
+with `#[reducer]`. 
+
+:::info Transactionality
+Crucially, **every reducer call executes within a single, atomic database transaction.** If the reducer function completes successfully (returns `()` or `Ok(())`), all database modifications made within it are committed together. If the reducer fails (panics or returns `Err(...)`), the transaction is aborted, and **all database changes made during that specific call are automatically rolled back**, ensuring data consistency.
+:::
+
+Reducers operate within a sandbox and have limitations:
+*   They cannot directly perform network I/O (e.g., using `std::net`).
+*   They cannot directly access the filesystem (e.g., using `std::fs` or `std::io`).
+*   External communication happens primarily through database table modifications (which clients can subscribe to) and logging (`log` crate).
+
+Reducers *can* call other reducers defined within the same module. This is a direct function call, not a network request, and executes within the same database transaction.
```

Comment:
This should be more explicit that there is no sub-transaction.

Reply:
Clarified!